test/unit/mem/buf/src/ConstBufTest.cpp

```diff
@@ -0,0 +1,119 @@
+/* Copyright 2025 Anton Reinhard
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+#include <alpaka/mem/buf/Traits.hpp>
+#include <alpaka/test/Extent.hpp>
+#include <alpaka/test/acc/TestAccs.hpp>
+#include <alpaka/test/mem/view/ViewTest.hpp>
+#include <alpaka/test/queue/Queue.hpp>
+
+#include <catch2/catch_message.hpp>
+#include <catch2/catch_template_test_macros.hpp>
+#include <catch2/catch_test_macros.hpp>
+
+#include <numeric>
+#include <type_traits>
+
+namespace buftest
+{
+    template<typename TDim, typename TDev, typename TElem, typename TIdx, typename TExtent>
+    auto allocConstBuf(TDev dev, TExtent extent)
+    {
+        return alpaka::makeConstBuf(alpaka::allocBuf<TElem, TIdx>(dev, extent));
+    }
+
+    template<typename TDim, typename TElem, typename TIdx, typename TExtent>
+    auto allocConstBuf(alpaka::DevCpu dev, TExtent extent) -> alpaka::ConstBufCpu<TElem, TDim, TIdx>
+    {
+        return alpaka::makeConstBuf(alpaka::allocBuf<TElem, TIdx>(dev, extent));
+    }
+
+#if defined(ALPAKA_ACC_GPU_CUDA_ENABLED)
+    template<typename TDim, typename TElem, typename TIdx, typename TExtent>
+    auto allocBuf(alpaka::DevCudaRt dev, TExtent extent) -> alpaka::ConstBufCudaRt<TElem, TDim, TIdx>
+    {
+        return alpaka::makeConstBuf(alpaka::allocBuf<TElem, TIdx>(dev, extent));
+    }
+#endif
+#if defined(ALPAKA_ACC_GPU_HIP_ENABLED)
+    template<typename TDim, typename TElem, typename TIdx, typename TExtent>
+    auto allocBuf(alpaka::DevHipRt dev, TExtent extent) -> alpaka::ConstBufHipRt<TElem, TDim, TIdx>
+    {
+        return alpaka::makeConstBuf(alpaka::allocBuf<TElem, TIdx>(dev, extent));
+    }
+#endif
+#if defined(ALPAKA_ACC_SYCL_ENABLED) and defined(ALPAKA_SYCL_ONEAPI_CPU)
+    template<typename TDim, typename TElem, typename TIdx, typename TExtent>
+    auto allocBuf(alpaka::DevCpuSycl dev, TExtent extent) -> alpaka::ConstBufCpuSycl<TElem, TDim, TIdx>
+    {
+        return alpaka::makeConstBuf(alpaka::allocBuf<TElem, TIdx>(dev, extent));
+    }
+#endif
+#if defined(ALPAKA_ACC_SYCL_ENABLED) and defined(ALPAKA_SYCL_ONEAPI_GPU)
+    template<typename TDim, typename TElem, typename TIdx, typename TExtent>
+    auto allocBuf(alpaka::DevGpuSyclIntel dev, TExtent extent) -> alpaka::ConstBufGpuSyclIntel<TElem, TDim, TIdx>
+    {
+        return alpaka::makeConstBuf(alpaka::allocBuf<TElem, TIdx>(dev, extent));
```

Comment:
I have currently problems following the PR and I am not sure anymore what this PR is trying to solve, maybe we should discuss this in our developer meeting.
I thought the `ConstBuff` should convert an exiting buffer into a nonmutable buffer, which makes IMO sense.
In this line we create a constant buffer that is not mutable, I am asking myself how to fill  nonmutable buffer with data :thinking:

Reply:
As I also explained in the meeting, this `allocBuf` function is only used for testing where I want to check that it is impossible to mute the ConstBuf's buffer. Therefore I don't care what data is inside.
The `makeConstBuf` function is simply a convenient explicit cast, creating a constant buffer from the mutable buffer. It just calls the copy constructor, but calling the copy constructor yourself would be tedious because you'd need to also provide all the correct template arguments for the resulting `ConstBuf`.